src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,54 @@
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpErrorResponse, HttpHeaders} from "@angular/common/http";
+import {SessionStorageService} from "./session-storage.service";
+import {BehaviorSubject, Observable} from "rxjs";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  private isAuthorized$$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  public isAuthorized$: Observable<boolean> = this.isAuthorized$$.asObservable();
+
+  sessionStorageService: SessionStorageService;
+
+  constructor(private http: HttpClient, private sessionStorage: SessionStorageService) {
+    this.sessionStorageService = sessionStorage;
+  }
```

Comment:
```suggestion
  constructor(private http: HttpClient, private sessionStorageService: SessionStorageService) {
  }
```
Extra field is not needed. Adding private or public modifier for arguments means that we define a new field in the class and can use it through `this`